### PR TITLE
Fixing issue where stable time would always be overwritten with utcnow()

### PR DESCRIPTION
**Pull request by kevgliss**
When updating the stable_at field, we were always setting this field to be utcnow(), this first checks for an existing stable_at before setting the field.

### CODE REVIEW

File: src/dispatch/incident/models.py [hunk 1]
```diff
@@ -104,7 +104,8 @@ def last_status_report(self):
     @staticmethod
     def _status_time(mapper, connection, target):
         if target.status == IncidentStatus.stable:
-            target.stable_at = datetime.utcnow()
+            if not target.stable_at:
```
Comment (kevgliss) [inline src/dispatch/incident/models.py]: @mvilanova this fixes the immediate issue, but I'm inclined to move this logic to the flows itself (this was a tricky one to track down). Thoughts?

Comment (mvilanova) [reply src/dispatch/incident/models.py]: Shouldn't `update` [here](https://github.com/Netflix/dispatch/blob/develop/src/dispatch/incident/views.py#L94) take care of updating the time?

File: src/dispatch/static/dispatch/src/incident/Table.vue [hunk 2]
```diff
@@ -46,9 +46,7 @@
                 <v-icon small class="mr-2" @click="createEditShow(item)">edit</v-icon>
               </template>
               <template v-slot:item.created_at="{ item }">
```
Comment (mvilanova) [inline src/dispatch/static/dispatch/src/incident/Table.vue]: I think this should be `item.reported_at`

File: src/dispatch/plugins/dispatch_slack/views.py [hunk 3]
```diff
@@ -509,12 +511,39 @@ def command_functions(command: str):
...
+    )
+    incident = incident_service.get(db_session=db_session, incident_id=incident_id)
+    existing_incident = IncidentRead.from_orm(incident)
+    incident_service.update(db_session=db_session, incident=incident, incident_in=incident_in)
+    incident_flows.incident_update_flow(user_email, incident_id, existing_incident, notify)
+
+
+@background_task
+def handle_assign_role_action(user_email, incident_id, action, db_session=None):
+    """Messages slack dialog daa into some thing that Dispatch can use."""
```
Comment (mvilanova) [inline src/dispatch/plugins/dispatch_slack/views.py]: `data`

File: src/dispatch/plugins/dispatch_slack/views.py [hunk 4]
```diff
@@ -509,12 +511,39 @@ def command_functions(command: str):
...
+    assignee_role = action["submission"]["role"]
+    assignee_email = get_user_email(client=slack_client, user_id=assignee_user_id)
+    incident_flows.incident_assign_role_flow(user_email, incident_id, assignee_email, assignee_role)
+
+
 def action_functions(action: str):
     """Interprets the action and routes it the appropriate function."""
     action_mappings = {
         SLACK_COMMAND_STATUS_REPORT_SLUG: [status_report_flows.new_status_report_flow],
         SLACK_COMMAND_ASSIGN_ROLE_SLUG: [incident_flows.incident_assign_role_flow],
```
Comment (mvilanova) [inline src/dispatch/plugins/dispatch_slack/views.py]: Shouldn't this command be mapped to `handle_assign_role_action`?

File: src/dispatch/plugins/dispatch_slack/views.py [hunk 5]
```diff
@@ -701,6 +732,7 @@ def verify_timestamp(timestamp: int):
 
     # We resolve the user's email
     user_id = action["user"]["id"]
+    print(user_id)
```
Comment (mvilanova) [inline src/dispatch/plugins/dispatch_slack/views.py]: We should remove this `print()`.

File: src/dispatch/incident/flows.py [hunk 6]
```diff
@@ -497,77 +497,21 @@ def incident_active_flow(incident_id: int, command: Optional[dict] = None, db_se
...
-                    {
-                        "type": "section",
-                        "text": {
-                            "type": "plain_text",
-                            "text": "The incident is already active. Aborting command...",
-                        },
-                    }
-                ],
-            )
-        return
```
Comment (mvilanova) [inline src/dispatch/incident/flows.py]: Are you planning to add this logic to a new function called, for example, `handle_mark_incident_acttive_command` in the views module of the Slack plugin?

File: src/dispatch/incident/flows.py [hunk 7]
```diff
@@ -497,77 +497,21 @@ def incident_active_flow(incident_id: int, command: Optional[dict] = None, db_se
...
-    log.debug(f"We have updated the status of the incident to {IncidentStatus.active}.")
-
     # we update the status of the external ticket
     update_incident_ticket(
         incident.ticket.resource_id,
         incident_type=incident.incident_type.name,
         status=IncidentStatus.active.lower(),
     )
 
-    log.debug(f"We have updated the status of the external ticket to {IncidentStatus.active}.")
```
Comment (mvilanova) [inline src/dispatch/incident/flows.py]: Any reason to not keep this one?

File: src/dispatch/incident/flows.py [hunk 8]
```diff
@@ -497,77 +497,21 @@ def incident_active_flow(incident_id: int, command: Optional[dict] = None, db_se
...
-                    {
-                        "type": "section",
-                        "text": {
-                            "type": "plain_text",
-                            "text": "The incident is already stable. Aborting command...",
-                        },
-                    }
-                ],
-            )
-            return
```
Comment (mvilanova) [inline src/dispatch/incident/flows.py]: Same comment as the one for the active flow.

File: src/dispatch/plugins/dispatch_slack/config.py [hunk 9]
```diff
@@ -29,8 +29,8 @@
 SLACK_COMMAND_ASSIGN_ROLE_SLUG = config(
     "SLACK_COMMAND_ASSIGN_ROLE_SLUG", default="/dispatch-assign-role"
 )
-SLACK_COMMAND_EDIT_INCIDENT_SLUG = config(
-    "SLACK_COMMAND_EDIT_INCIDENT_SLUG", default="/dispatch-edit-incident"
+SLACK_COMMAND_UPDATE_INCIDENT_SLUG = config(
```
Comment (mvilanova) [inline src/dispatch/plugins/dispatch_slack/config.py]: Note to update command name in the Slack UI, and Dispatch documentation.

File: src/dispatch/incident/flows.py [hunk 10]
```diff
@@ -497,77 +497,21 @@ def incident_active_flow(incident_id: int, command: Optional[dict] = None, db_se
...
-    log.debug(f"We have updated the status of the external ticket to {IncidentStatus.active}.")
-
-    # we update the conversation topic
-    set_conversation_topic(incident)
-
-    # we send the active notifications
-    send_incident_status_notifications(incident, db_session)
-
-    log.debug("We have sent the incident active notifications.")
-
```
Comment (mvilanova) [inline src/dispatch/incident/flows.py]: When are we calling these functions when we call this flow directly from the Slack code when we run `/dispatch-mark-active`?

File: src/dispatch/incident/flows.py [hunk 11]
```diff
@@ -584,9 +528,6 @@ def incident_stable_flow(incident_id: int, command: Optional[dict] = None, db_se
 
     log.debug(f"We have updated the status of the external ticket to {IncidentStatus.stable}.")
 
-    # we update the conversation topic
-    set_conversation_topic(incident)
-
```
Comment (mvilanova) [inline src/dispatch/incident/flows.py]: Where are we updating the topic now? We also call this flow directly from the Slack code when run `/dispatch-mark-stable`.

File: src/dispatch/incident/flows.py [hunk 12]
```diff
@@ -676,27 +614,7 @@ def incident_closed_flow(incident_id: int, command: Optional[dict] = None, db_se
...
-                command["user_id"],
-                "Mark Incident Stable Notification",
-                blocks=[
-                    {
-                        "type": "section",
-                        "text": {"type": "plain_text", "text": "Marking the incident as stable..."},
-                    }
-                ],
-            )
-        incident_stable_flow(incident_id, command=command, db_session=db_session)
```
Comment (mvilanova) [inline src/dispatch/incident/flows.py]: Same as active and stable flows, plus we need to run the stable flow if incident status is active.

File: src/dispatch/incident/flows.py (hunk 10 above)
Comment (kevgliss) [reply src/dispatch/incident/flows.py]: These are now handled by the `update` command.

File: src/dispatch/incident/flows.py (hunk 11 above)
Comment (kevgliss) [reply src/dispatch/incident/flows.py]: Same as above.

File: src/dispatch/incident/flows.py (hunk 6 above)
Comment (kevgliss) [reply src/dispatch/incident/flows.py]: Yes, we could add it to the handle_mark_incident_active, or we could centralize it based on a signal/source.

File: src/dispatch/incident/flows.py (hunk 7 above)
Comment (kevgliss) [reply src/dispatch/incident/flows.py]: No, I will put it back in.

File: src/dispatch/incident/flows.py (hunk 10 above)
Comment (mvilanova) [reply src/dispatch/incident/flows.py]: We are not calling the `update_flow` function when we run the command. Same for the stable and closed ones. See https://github.com/Netflix/dispatch/blob/develop/src/dispatch/plugins/dispatch_slack/views.py#L504

Comment (mvilanova) [reply src/dispatch/incident/flows.py]: We could add status to the update dialog and get rid of the mark [active|stable|closed] commands like we've discussed in the past.

File: src/dispatch/incident/flows.py (hunk 12 above)
Comment (kevgliss) [reply src/dispatch/incident/flows.py]: Same as others stable flow should run if `update_flow` thinks it needs to.

**Pull request merged**